### PR TITLE
[fix]: try to reduce escapes to heap

### PR DESCRIPTION
**Pull request by SamYuan1990**
https://github.com/sustainable-computing-io/kepler/actions/runs/12177539929/job/33966954289
seems we have many nits can be improved to reduce our memory escapes to heap case.
in this PR, try to improve parts of them.

### CODE REVIEW

File: pkg/collector/stats/types/types.go [hunk 1]
```diff
@@ -14,10 +14,10 @@ type UInt64Stat struct {
 }
 
 func NewUInt64Stat(aggr, delta uint64) *UInt64Stat {
-	stat := &UInt64Stat{}
+	stat := UInt64Stat{}
```
Comment (SamYuan1990) [inline pkg/collector/stats/types/types.go:17]: fix `Error: pkg/collector/stats/types/types.go:17:10: &UInt64Stat{} escapes to heap`
ref https://github.com/sustainable-computing-io/kepler/actions/runs/12177539929/job/33966954289
and
https://github.com/sustainable-computing-io/kepler/actions/runs/12197388859/job/34026929446?pr=1886

File: pkg/config/config.go [hunk 2]
```diff
@@ -234,7 +234,7 @@ func getBoolConfig(configKey string, defaultBool bool) bool {
 }
 
 func getIntConfig(configKey string, defaultInt int) int {
-	defaultValue := fmt.Sprintf("%d", defaultInt)
+	defaultValue := strconv.Itoa(defaultInt)
```
Comment (SamYuan1990) [inline pkg/config/config.go:237]: fix `Error: pkg/config/config.go:237:36: defaultInt escapes to heap`
ref https://github.com/sustainable-computing-io/kepler/actions/runs/12177539929/job/33966954289 and https://github.com/sustainable-computing-io/kepler/actions/runs/12197388859/job/34026929446?pr=1886

File: pkg/kubelet/kubelet_pod_lister.go [hunk 3]
```diff
@@ -50,6 +51,8 @@ func init() {
 		port = "10250"
 	}
 	podURL = "https://" + nodeName + ":" + port + "/pods"
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	client = http.Client{}
```
Comment (SamYuan1990) [inline pkg/kubelet/kubelet_pod_lister.go:55]: `Error: pkg/kubelet/kubelet_pod_lister.go:69:12: &http.Client{} escapes to heap`
make http client reuse

**Review by rootfs — APPROVED** [review]

**Comment by rootfs** [comment]
lgtm, @vimalk78 can you take a final check? thanks

**Pull request merged**